### PR TITLE
feat: Add CloudQuery integration (ECOINT-11)

### PR DESCRIPTION
**Pull request by erezrokah**
### What does this PR do?

Add [CloudQuery](https://www.cloudquery.io/) integration

### Motivation

Get listed on the integrations page per https://docs.datadoghq.com/developers/integrations/?tab=integrations

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [x] If this PR includes a log pipeline, please add a description describing the remappers and processors. 

### Additional Notes

Anything else we should know when reviewing?

I followed the guide in https://docs.datadoghq.com/developers/integrations/?tab=integrations as best as I can. Feedback welcomed!

### CODE REVIEW

**Comment by drichards-87** [comment]
Created Jira card for Docs Team review.

File: cloudquery/README.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    -e DATADOG_SITE=$DATADOG_SITE \
+    -e DATADOG_API_KEY=$DATADOG_API_KEY \
+    --hostname $(hostname) \
+    -v $(pwd)/otel_collector_config.yaml:/etc/otelcol-contrib/config.yaml \
+    otel/opentelemetry-collector-contrib:0.104.0
+```
+
+> For additional ways to run the collector, please refer to the [official documentation][7].
+
+- Direct OTEL Ingestion by the Datadog Agent via a configuration file
```
Comment (estherk15) [inline cloudquery/README.md:76]:
```suggestion
#### Datadog Agent OTEL ingestion through a configuration file
```

File: cloudquery/README.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+[Locate][8] your `datadog.yaml` file and add the following configuration:
```
Comment (estherk15) [inline cloudquery/README.md:78]:
```suggestion
1. Locate your [`datadog.yaml` Agent configuration file][8] and add the following configuration:
```

File: cloudquery/README.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  receiver:
+    protocols:
+      http:
+        endpoint: 0.0.0.0:4318
+  logs:
+    enabled: true
+logs_enabled: true
+```
+
+[Restart][9] the Datadog agent for the change to take effect.
```
Comment (estherk15) [inline cloudquery/README.md:91]:
```suggestion
1. [Restart][9] the Datadog agent for the change to take effect.
```

File: cloudquery/README.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+- Direct OTEL ingestion by the Datadog Agent via environment variables
```
Comment (estherk15) [inline cloudquery/README.md:93]:
```suggestion
#### Datadog Agent OTEL ingestion through environment variables
```

File: cloudquery/README.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+Pass the `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT` environment variable to the Datadog agent with a value of `0.0.0.0:4318`.
```
Comment (estherk15) [inline cloudquery/README.md:95]:
```suggestion
1. Pass the `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT` environment variable to the Datadog Agent with a value of `0.0.0.0:4318`.
```

File: cloudquery/README.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+If you're using Docker compose, you can find an example below:
```
Comment (estherk15) [inline cloudquery/README.md:96]:
```suggestion
If you're using Docker compose, see the example below:
```

File: cloudquery/README.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+[1]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/cloudquery/images/cloudquery_logo_png_dark_background.png
+[2]: https://www.cloudquery.io/
+[3]: https://github.com/cloudquery/cloudquery
+[4]: https://www.cloudquery.io/pricing
+[5]: https://opentelemetry.io/
+[6]: https://docs.datadoghq.com/opentelemetry/
+[7]: https://docs.datadoghq.com/opentelemetry/collector_exporter/deployment#running-the-collector
+[8]: https://docs.datadoghq.com/agent/configuration/agent-configuration-files/
+[9]: https://docs.datadoghq.com/agent/configuration/agent-commands/#restart-the-agent
+[10]: https://docs.datadoghq.com/agent/configuration/agent-commands/#restart-the-agent
```
Comment (estherk15) [inline cloudquery/README.md:176]: duplicate

```suggestion
```

File: cloudquery/README.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+      DD_API_KEY: redacted
+      DD_SITE: "datadoghq.eu"
+      DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT: "0.0.0.0:4318"
+      DD_LOGS_ENABLED: "true"
+      DD_OTLP_CONFIG_LOGS_ENABLED: "true"
+    ports:
+      - "4318:4318"
+```
+
+[Restart][10] the Datadog agent for the change to take effect.
```
Comment (estherk15) [inline cloudquery/README.md:117]:
```suggestion
1. [Restart][9] the Datadog agent for the change to take effect.
```

File: cloudquery/README.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+
+> For additional ways to configure the Datadog agent, please refer to the [official documentation][11].
```
Comment (estherk15) [inline cloudquery/README.md:119]:
```suggestion
For more ways to configure the Datadog Agent, see [OTLP Ingestion by the Datadog Agent][11].
```

File: cloudquery/README.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+Once you have the agent or collector ready, you can specify the endpoint in the source spec:
```
Comment (estherk15) [inline cloudquery/README.md:121]: Is this optional? If not, recommend moving this to each configuration as the final step. For example: 
```
1. Restart the Datadog agent for the change to take effect.
2. After you have the Agent ready, specify the endpoint in the source spec:
```
```suggestion
After you have the Agent or collector ready, specify the endpoint in the source spec:
```

File: cloudquery/README.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+  destinations: ["postgresql"]
+  otel_endpoint: "0.0.0.0:4318"
+  otel_endpoint_insecure: true
+  spec:
+```
+
+### Validation
+
+Run `cloudquery sync spec.yml`.
+Once ingestion starts you should be able to start seeing the traces in Datadog under APM->Traces->Explorer.
```
Comment (estherk15) [inline cloudquery/README.md:140]:
```suggestion
After ingestion starts, you should start seeing the traces in the Datadog [**APM Traces Explorer**][13].
```

File: cloudquery/README.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+You should see metrics and logs in the respective sections of the Datadog UI as well.
```
Comment (estherk15) [inline cloudquery/README.md:141]:
```suggestion
You can also validate metrics and logs in the [**Metrics Summary**][14] and [**Log Explorer**][15].
```

File: cloudquery/README.md [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+The CloudQuery does not include any service checks.
+
+### Events
+
+The CloudQuery does not include any events.
+
+## Uninstallation
+
+If you use the OpenTelemetry collector, you can stop it by running `docker stop <container_id>`.
+IF you use the Datadog agent, remove the configuration or environment variables you added and [restart][10] the agent.
```
Comment (estherk15) [inline cloudquery/README.md:160]:
```suggestion
If you use the Datadog Agent, remove the configuration or environment variables you added and [restart][9] the agent.
```

File: cloudquery/README.md [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+Finally, delete the dashboard from your Datadog account.
+
+## Support
+
+Need help? [Contact us][4].
```
Comment (estherk15) [inline cloudquery/README.md:165]:
```suggestion
For support, contact [CloudQuery][4].
```

File: cloudquery/assets/dashboards/cloudquery_syncs_insights.json [hunk 15]
```diff
@@ -0,0 +1,324 @@
...
+              "tick_edge": "left",
+              "has_padding": true
+            },
+            "layout": { "x": 0, "y": 0, "width": 6, "height": 2 }
+          },
+          {
+            "id": 1803149444713820,
+            "definition": {
+              "type": "note",
+              "content": "Use this dashboard to get insights into CloudQuery syncs and optimize them. For more information on setting up CloudQuery with Datadog visit our [docs](https://docs.cloudquery.io/docs/advanced-topics/monitoring#opentelemetry-and-datadog).",
```
Comment (estherk15) [inline cloudquery/assets/dashboards/cloudquery_syncs_insights.json:34]:
```suggestion
              "content": "Use this dashboard to get insights into CloudQuery syncs and optimize them. For more information on setting up CloudQuery with Datadog, see the [CloudQuery documentation](https://docs.cloudquery.io/docs/advanced-topics/monitoring#opentelemetry-and-datadog).",
```

**Review by estherk15 — COMMENTED** [review]
Left some suggestions for documentation, let me know if you have any questions or would like a re-review!

File: cloudquery/README.md [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+### Installation
+
+Install the Datadog Agent version >=6.48.0 or >=7.48.0, which supports ingesting OpenTelemetry traces, metrics, and logs from CloudQuery.
+Alternatively, you can use OpenTelemetry Collector and Datadog Exporter as described below.
+
+### Configuration
+
+CloudQuery supports [OpenTelemetry][5] traces, metrics and logs out of the box.
+There are multiple ways to configure OpenTelemetry with Datadog. We'll show only a subset of them here, and you can find more information [in the Datadog docs][6].
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply cloudquery/README.md:21]: Great suggestion, wasn't sure if I can use more nesting for the headers (e.g. `####`)

File: cloudquery/README.md (hunk 10 above)
Comment (erezrokah) [reply cloudquery/README.md:127]: This is required, good suggestion 👍

**Comment by erezrokah** [comment]
Thanks for the review @estherk15, great suggestions 🚀 I also followed up with https://github.com/DataDog/integrations-extras/pull/2451/commits/087f84ae9d7591c359601eb8629ba28d536e5c65 and https://github.com/DataDog/integrations-extras/pull/2451/commits/06a21eef487b51742391b4bad2a1911624719eb5 to fix the links and lists numbering (as the lists numbering is not preserved after the code blocks).

File: cloudquery/README.md [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+```bash
+docker run \
+    -p 4318:4318 \
+    -e DATADOG_SITE=$DATADOG_SITE \
+    -e DATADOG_API_KEY=$DATADOG_API_KEY \
+    --hostname $(hostname) \
+    -v $(pwd)/otel_collector_config.yaml:/etc/otelcol-contrib/config.yaml \
+    otel/opentelemetry-collector-contrib:0.104.0
+```
+
```
Comment (estherk15) [inline cloudquery/README.md:81]:
````suggestion
3. After you have the collector ready, specify the endpoint in the source spec:

```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  registry: "cloudquery"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
  otel_endpoint: "0.0.0.0:4318"
  otel_endpoint_insecure: true
  spec:
```

````

File: cloudquery/README.md [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+  receiver:
+    protocols:
+      http:
+        endpoint: 0.0.0.0:4318
+  logs:
+    enabled: true
+logs_enabled: true
+```
+
+2. [Restart][9] the Datadog agent for the change to take effect.
```
Comment (estherk15) [inline cloudquery/README.md:99]:
````suggestion
2. [Restart][9] the Datadog agent for the change to take effect.

3. After you have the Agent ready, specify the endpoint in the source spec:

```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  registry: "cloudquery"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
  otel_endpoint: "0.0.0.0:4318"
  otel_endpoint_insecure: true
  spec:
```
````

File: cloudquery/README.md [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+      DD_API_KEY: redacted
+      DD_SITE: "datadoghq.eu"
+      DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT: "0.0.0.0:4318"
+      DD_LOGS_ENABLED: "true"
+      DD_OTLP_CONFIG_LOGS_ENABLED: "true"
+    ports:
+      - "4318:4318"
+```
+
+2. [Restart][9] the Datadog agent for the change to take effect.
```
Comment (estherk15) [inline cloudquery/README.md:125]:
````suggestion
2. [Restart][9] the Datadog agent for the change to take effect.

3. After you have the Agent ready, specify the endpoint in the source spec:
```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  registry: "cloudquery"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
  otel_endpoint: "0.0.0.0:4318"
  otel_endpoint_insecure: true
  spec:
```
````

File: cloudquery/README.md [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+  name: "aws"
+  path: "cloudquery/aws"
+  registry: "cloudquery"
+  version: "VERSION_SOURCE_AWS"
+  tables: ["aws_s3_buckets"]
+  destinations: ["postgresql"]
+  otel_endpoint: "0.0.0.0:4318"
+  otel_endpoint_insecure: true
+  spec:
+```
```
Comment (estherk15) [inline cloudquery/README.md:143]:
```suggestion
```

**Review by estherk15 — APPROVED** [review]
Thanks for taking my suggestions! Looks great!

**Comment by erezrokah** [comment]
> Thanks for taking my suggestions! Looks great!

Thanks for reviewing, I added another commit in https://github.com/DataDog/integrations-extras/pull/2451/commits/ac03e1c416792a36896ec93b7dd3b971cc88cc47 to better explain the version placeholders. I also removed the `registry: cloudquery` as that's the default and optional in our configuration

**Review by estherk15 — APPROVED** [review]

**Comment by erezrokah** [comment]
Thanks @estherk15, per a meeting with the team in charge of partner integrations, we'll submit this as a community integration until there's official OTEL support

**Comment by erezrokah** [comment]
> we'll submit this as a community integration until there's official OTEL support

OK so I guess I got confused by something, since the instructions in https://docs.datadoghq.com/agent/guide/use-community-integrations/?tab=agentv721v621 say to open a PR to this repo.

Can we accept this PR as a community integration?

**Comment by erezrokah** [comment]
Moving to draft until I fix the log pipelines CI

**Comment by erezrokah** [comment]
Not sure how to fix these errors:
image

This is how I configured the filter in the Datadog UI for AWS:
image

We send a `service` field with `name` and `version` properties in the logs so we should be able to filter by those no?

**Comment by erezrokah** [comment]
OK, removed the log pipelines in the initial version

**Comment by erezrokah** [comment]
https://github.com/DataDog/integrations-extras/pull/2451/commits/14209ecb539754e70796c5651d4f271b4f85e072 marks this as a community integration

**Review by HadhemiDD — APPROVED** [review]

**Comment by HadhemiDD** [comment]
Can you please fix the conflict in CODEOWNERS file? you just need to rebase it with master as it has conflicts with it.

**Comment by erezrokah** [comment]
> Can you please fix the conflict in CODEOWNERS file? you just need to rebase it with master as it has conflicts with it.

Done! Thanks for the bump

**Pull request merged**